### PR TITLE
chore: make sure that version bumping happens everytime

### PR DESCRIPTION
**Pull request by jakobmoellerdev**
#### What this PR does / why we need it

This makes sure that we dont only do a patch bump on main after a successful branch cut, but we also do a bump on the release branch (for the next z / patch version) after the minor has been released.

Creates a shared workflow that can be called from other workflows.

#### Which issue(s) this PR fixes

Ensures more automated workflows after the rework from https://github.com/open-component-model/ocm/issues/995

Example run at https://github.com/open-component-model/ocm-cicd-playground/actions/runs/11838213132/job/32986884073

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -194,7 +194,7 @@ jobs:
       run: |
         set -e
         git checkout ${GITHUB_REF#refs/heads/}
-        v="$(go run ./api/version/generate bump-version)"
+        v="$(go run ./api/version/generate bump-minor)"
```
Comment (hilmarf) [inline .github/workflows/release.yaml:197]: hmm... what happens when we need to create a patch-release (e.g. `v0.17.1`)? Then this is executed on `releases/vX.Y` branch right? Or am I wrong? Wouldn't then the version be bumped to `v0.18.x` ?

**Review by hilmarf — DISMISSED** [review]

**Review by frewilhelm — DISMISSED** [review]

**Review by frewilhelm — APPROVED** [review]
lgtm (even though I have to admit that this whole generate-version scripts/programs are really confusing)

**Comment by jakobmoellerdev** [comment]
Understood. If you could let me know how we can improve them, please tell me straight away. I will try to simplify as much as possible but currently my PRs are focusing on only "touching up" without rewriting

**Comment by frewilhelm** [comment]
I created an issue https://github.com/open-component-model/ocm/issues/1094. Although I agree, that this has no priority.

**Pull request merged**